src/sample/Program.cs

```diff
@@ -102,18 +105,35 @@ static CommandLineBuilder BuildCommandLine()
 
             var builder = new CommandLineBuilder(rootCommand);
             var debugOption = new Option<bool>("--debug", "Enable debug output");
+            var headersOption = new Option<string[]>("--headers", "Add custom request headers to the request. Can be used multiple times to add many headers. e.g. --headers key1=value1 --headers key2=value2");
+            headersOption.Arity = ArgumentArity.ZeroOrMore;
 
-            builder.AddMiddleware(async (ic, next) =>
-            {
-                debugEnabled = ic.ParseResult.GetValueForOption<bool>(debugOption);
-                if (debugEnabled)
+            static void AddOptionToCommandIf(ref Command command, in Option option, Func<Command, bool> predicate) {
+                if (predicate(command)) {
+                    command.AddOption(option);
+                }
+
+                foreach (var sym in command.Children)
```

Comment:
```suggestion
                foreach (var sym in command.Subcommands)
```
Then drop the if statement?